trackio/ui.py

```diff
@@ -139,9 +142,53 @@ def toggle_timer(cb_value):
         return gr.Timer(active=False)
 
 
-def log(project: str, run: str, metrics: dict[str, Any], dataset_id: str) -> None:
-    # Note: the type hint for dataset_id should be str | None but gr.api
+def log(
+    project: str, run: str, metrics: dict[str, Any], dataset_id: str, hf_token: str
+) -> None:
+    # Note: the type hint for dataset_id and hf_token should be str | None but gr.api
     # doesn't support that, see: https://github.com/gradio-app/gradio/issues/11175#issuecomment-2920203317
```

Comment:
Unrelated, but in the latest version of `gradio`, we've fixed this, so now we can update the type hints to be `str | None`.